### PR TITLE
Allow local checkout environment with sidekick on bold.ninja tunnels

### PR DESCRIPTION
**Pull request by cxfos**
This is a development quality of life change, that allow to use the calls to bold.ninja domains for local developers without the need to manually change the code to use these domains.

### CODE REVIEW

File: Model/Http/BoldClient.php [hunk 1]
```diff
@@ -152,17 +144,32 @@ private function getHeaders(int $websiteId): array
...
+            $path = $parseApiUrl['path'];
+            $tunnelDomain = ltrim($path, '/');
+            $baseApiUrl = $scheme.'://'.$host;
+
+            if ($url === 'shops/v1/info') {
+                $apiUrl = $baseApiUrl;
+            }
+
+            if (str_contains($url, 'checkout_sidekick')) {
+                $apiUrl = $baseApiUrl.'/sidekick-'.$tunnelDomain;
```
Comment (BoldBrandonM) [inline Model/Http/BoldClient.php:165]: This would only work for specific tunnel names which are structured a certain way. I'm not against it for a local solution, but we should add that to the readme most likely.

**Review by BoldBrandonM — APPROVED** [review]

File: Model/Http/BoldClient.php (hunk 1 above)
Comment (cxfos) [reply Model/Http/BoldClient.php:165]: Yes, I was thinking on that exactly when doing this. 
Since you are here, could you let me know if we have a readme for the sidekick that we instruct on the tunnel creation?

Comment (BoldBrandonM) [reply Model/Http/BoldClient.php:165]: We do have a readme right in the project root for Sidekick, so any specific instructions about how set things up could go in there. There's also a `how_to_install.md` file which includes instructions for required setup steps so if you feel it fits better there that works too.

File: Model/Http/BoldClient.php
Comment (JosephLeedy) [inline Model/Http/BoldClient.php:1]: Should the formatting and style changes made by this commit be separated into their own PR as they're unrelated?

Comment (cxfos) [reply Model/Http/BoldClient.php:1]: They were automatically done by the IDE. 

Since tha actual change is small and the style changes are simple, 1 file (the one related to the intended change), and not affecting any logic I don't think having a separated PR is justifyable, would just create an unecessary second round of reviews.

Comment (cxfos) [reply Model/Http/BoldClient.php:1]: I can just remove all the style changes too, and if at somepoint that becomes a priority on the pipeline we handle that on the whole project at once (I don't think it will happen tho).

Comment (JosephLeedy) [reply Model/Http/BoldClient.php:1]: In my opinion, the style changes should be done separately because they add noise that obscures the real changes made by this PR.

Comment (cxfos) [reply Model/Http/BoldClient.php:1]: As I mentioned is pretty small changes and quick scroll read changes, but I will remove.

I just want to mentioned that the work to remove it is bigger than any extra "noise" it causes given its size and time.

Comment (cxfos) [reply Model/Http/BoldClient.php:1]: Done.

File: Model/Http/BoldClient.php (hunk 1 above)
Comment (cxfos) [reply Model/Http/BoldClient.php:177]: @BoldBrandonM created a documentation PR for sidekick: https://github.com/bold-commerce/checkout-sidekick/pull/76

**Review by JosephLeedy — APPROVED** [review]
